include/boost/math/statistics/univariate_statistics.hpp

```diff
@@ -512,6 +513,58 @@ inline auto interquartile_range(RandomAccessContainer & v)
     return interquartile_range(v.begin(), v.end());
 }
 
+template<class RandomAccessIterator>
+auto mode(RandomAccessIterator first, RandomAccessIterator last)
+{
+    using Real = typename std::iterator_traits<RandomAccessIterator>::value_type;
+    using Size = typename std::iterator_traits<RandomAccessIterator>::difference_type;
+
+    std::vector<Real> sorted(first, last);
+    std::sort(sorted.begin(), sorted.end());
+
+    std::vector<Real> modes {};
+    Size max_counter {};
+    Real previous_search {boost::math::tools::max_value<Real>()};
+
+    if(sorted.front() == sorted.back())
+    {
+        modes.push_back(sorted.front());
+        return modes;
+    }
+
+    for(const Real i : sorted)
+    {
+        if(i == previous_search)
+        {
+            continue;
+        }
+        
+        Size current_count {std::count(sorted.begin(), sorted.end(), i)};
+        if(current_count > max_counter)
+        {
+            modes.clear();
```

Comment:
Isn't this call to clear pretty expensive?

Reply:
At worst it is linear, and should never be very large. Libstdc++ has it optimized to constant complexity for POD.

Reply:
+1.  Clear should be as simple as setting the size of the vector to zero for POD types.  For non POD types the destructors would all be called.  I guess a more streamlined version in that case would be to resize the vector to 1, and then assign the new value?